### ISSUE TITLE
Timeouts with AWS Elasticache

### ISSUE THREAD

**Issue by nathankodilla** [StackExchange/StackExchange.Redis#818]
We commonly get timeouts when using AWS Elasticache but are not hitting any kind of resource limits. However these timeouts only happen with the sync calls. If we switch the same function to use the Async function and await the timeouts will never happen. On our connection string we have keepAlive=15 and set PreserveAsyncOrder to false.

StackExchange.Redis.RedisTimeoutException: Timeout performing EXEC, inst: 0, queue: 135, qu: 116, qs: 19, qc: 0, wr: 1, wq: 1, in: 0, ar: 0

**Comment by jpjohnson** [comment]
Have you tried Increasing the number of threads using ThreadPool.SetMin ?

**Comment by mgravell** [comment]
The threadpool mind count shouldn't be related. I don't advise changing
that.

The timeout for async works differently - there's an open feature request
to improve that. I guess the question becomes: what was happening at the
time? Is there anything in the slowlog? Were you sending / receiving any
huge blobs, etc?

**Comment by NickCraver** [comment]
@nathankodilla Do you have any info to add here? Without some more context, these things are impossible to diagnose with so many pieces involved in the pipe.

**Comment by nathankodilla** [comment]
We have tested several solutions and what did work was increasing the thread pool min value to around 1000. This number does have to be increased as our application grows and the more concurrent load we add to the app. There were no slowlogs that were occurring and our json blobs were small in size. Again if we used async for everything it works fine, only sync calls were an issue and one library we use only does sync calls.
